tests/helpers.bash

```diff
@@ -568,3 +574,125 @@ function stop_git_daemon() {
     rm -f ${TESTDIR}/git-daemon/pid
   fi
 }
+
+# Bring up a registry server using buildah with vfs and chroot as a cheap
+# substitute for podman, accessible only to user $1 using password $2 on the
+# local system at a dynamically-allocated port.
+# Requires openssl.
+# If a user name and password are supplied as the two parameters, htpasswd
+# should be available so that they can be hashed.
+# Sets REGISTRY_PID, REGISTRY_PORT (to append to "localhost:"), and
+# REGISTRY_DIR (where the CA cert can be found) on success.
+function start_registry() {
+  local REGISTRY_IMAGE=quay.io/libpod/registry:2.8
+  local config='
+version: 0.1
+log:
+  fields:
+    service: registry
+storage:
+  cache:
+    blobdescriptor: inmemory
+  filesystem:
+    rootdirectory: /var/lib/registry
+http:
+  addr: :0
+  headers:
+    X-Content-Type-Options: [nosniff]
+  tls:
+    certificate: /etc/docker/registry/localhost.crt
+    key: /etc/docker/registry/localhost.key
+health:
+  storagedriver:
+    enabled: true
+    interval: 10s
+    threshold: 3
+auth:
+  htpasswd:
+    realm: buildah-realm
+    path: /etc/docker/registry/htpasswd
+'
+  # use the result of "htpasswd -nbB testuser testpassword" if an alternate
+  # username and password are not specified
+  local htpasswd='testuser:$2y$05$OW6Qlf1ygRuqJxt/CYcBq.0MZARYLIvr.mgjksw2m7K4cwWkw3Pda'
+  if test "$1" = testuserfoo && test "$2" = testpassword ; then
+    htpasswd='testuserfoo:$2y$05$Pkly/9sA8iGY..2SaerFH.wr4qGPPaGZ66KMrvVmcCP8gaRPpeoeC'
+  elif test "${1:-testuser}" != testuser && test "${2:-testpassword}" != testpassword ; then
+    htpasswd=$(htpasswd -nbB "$1" "$2")
+    if test $? -ne 0 ; then
+      echo error computing hashed password
+      exit 1
```

Comment:
How about `die "Error computing hashed password"`?

Reply:
That'll work.  Changing it.